packages/tldraw/src/lib/ui/context/a11y.tsx

```diff
@@ -0,0 +1,60 @@
+import { Atom, useAtom } from '@tldraw/editor'
+import { createContext, useContext, useMemo } from 'react'
+
+/** @public */
+export type A11yPriority = 'polite' | 'assertive'
+
+/** @public */
+export interface TLUiA11y {
+	msg: string | undefined
+	priority?: A11yPriority
+}
+
+/** @public */
+export interface TLUiA11yContextType {
+	setMessage(msg: TLUiA11y): void
+	currentMsg: Atom<TLUiA11y>
+}
+
+/** @internal */
+export const A11yContext = createContext<TLUiA11yContextType | null>(null)
+
+/** @public */
+export interface A11yProviderProps {
+	children: React.ReactNode
+}
+
+/** @public @react */
+export function TldrawUiA11yProvider({ children }: A11yProviderProps) {
+	const currentMsg = useAtom<TLUiA11y>('a11y', { msg: '', priority: 'polite' })
+	const ctx = useContext(A11yContext)
+
+	const current = useMemo(
+		() => ({
+			currentMsg,
+			setMessage(msg: TLUiA11y) {
```

Comment:
maybe `announce` instead of `setMessage`? To me `setMessage` doesn't imply that then might be immediately reflected to the user

Reply:
good call 👍